tests/qa-activity-report/generate-qa-activity-report.go

```diff
@@ -0,0 +1,227 @@
+	daysStr := os.Getenv("DAYS")
+
+	if monthStr != "" && yearStr != "" {
+		month, err := strconv.Atoi(monthStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid month: %w", err)
+		}
+		year, err := strconv.Atoi(yearStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid year: %w", err)
+		}
+		startDate := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
+		endDate := startDate.AddDate(0, 1, -1)
+		since = startDate.Format("2006-01-02")
+		until = endDate.Format("2006-01-02")
+		period = fmt.Sprintf("%02d_%04d", month, year)
+	} else if daysStr != "" {
+		days, err := strconv.Atoi(daysStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid number of days: %w", err)
+		}
+		until = time.Now().Format("2006-01-02")
+		since = time.Now().AddDate(0, 0, -days).Format("2006-01-02")
+		period = fmt.Sprintf("Last_%d_days", days)
+	} else {
+		return "", "", "", fmt.Errorf("please provide either MONTH and YEAR or DAYS")
+	}
+
+	return since, until, period, nil
+}
+
+func getGitLog(since, until string) ([]string, error) {
+	cmd := exec.Command("git", "log", "--since="+since, "--until="+until, "--pretty=format:%H %ad", "--date=short", "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func getGitDiff(commit string) ([]string, error) {
+	cmd := exec.Command("git", "diff", commit+"~1", commit, "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func countAddedTestsInExamples(diffLines []string, startIndex int) int {
+	var inExamples bool
+	inExamples = false
+	addedTests := 0
+
+	for j := startIndex + 1; j < len(diffLines); j++ {
+		exampleLine := diffLines[j]
+		if strings.HasPrefix(exampleLine, "+") && strings.Contains(exampleLine, "Examples:") {
+			inExamples = true
+			continue
+		} else if inExamples {
+			trimmedLine := strings.TrimSpace(exampleLine)
+			if strings.HasPrefix(trimmedLine, "+") && strings.Contains(trimmedLine, "|") {
+				// Count a string if it starts with "+" and contains "|"
+				addedTests++
+			} else if strings.TrimSpace(exampleLine) == "" || !strings.HasPrefix(trimmedLine, "+") || !strings.HasPrefix(trimmedLine, "|") {
+				// Abort counting when a row that does not belong to the table is encountered
+				break
+			}
+		}
+	}
+
+	// We have one line | resource | which is not a test line. So we deleted one line from addedTest
+	// Examples:
+	// 	| resource      |
+	// 	| testfile.txt  |
+	// 	| FolderToShare |
+	if inExamples {
+		addedTests--
+	}
+	return addedTests
+}
+
+func checkChangedTests(diffLines []string, startIndex int) (bool, int) {
+	var inScenarios bool
+	inScenarios = false
```

Comment:
```suggestion
	var inScenarios bool
```
Same here. `bool` values always default to `false`